[RHEL-8.10] TestsAuthConfig.test_smartcard_auth fails with CDN content

The `TestsAuthConfig.test_smartcard_auth` test is failing in osbuild-composer RHEL-8.10 GA pipelines: 

- https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/9857133616
- https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/9857163531

In my opinion, this is caused by a change in the RPM that ships the config in RHEL-8, which falls outside the responsibility of the Image Builder team. The test case likely needs to be adjusted to reflect the current state of the configuration shipped in the RHEL-8.10 RPMs.

```
=================================== FAILURES ===================================
_ TestsAuthConfig.test_smartcard_auth[paramiko://civ-vm-image-rhel-810-x86-64-main-9857133616-11036.westeurope.cloudapp.azure.com] _
self = <test_generic.TestsAuthConfig object at 0x7f6ebc160670>
host = <testinfra.host.Host paramiko://civ-vm-image-rhel-810-x86-64-main-9857133616-11036.westeurope.cloudapp.azure.com>
    @pytest.mark.exclude_on(['>=rhel10.0'])
    def test_smartcard_auth(self, host):
        """
        Check file /etc/pam.d/smartcard-auth
        Bugzilla: 1983683
        """
    
>       self.__check_pam_d_file_content(host, 'smartcard-auth')
test_suite/generic/test_generic.py:941: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <test_generic.TestsAuthConfig object at 0x7f6ebc160670>
host = <testinfra.host.Host paramiko://civ-vm-image-rhel-810-x86-64-main-9857133616-11036.westeurope.cloudapp.azure.com>
file_name = 'smartcard-auth'
    def __check_pam_d_file_content(self, host, file_name):
        system_release_major_version = version.parse(host.system_info.release).major
        local_file = f'data/generic/{file_name}_rhel{system_release_major_version}'
        file_to_check = f'/etc/pam.d/{file_name}'
    
>       assert test_lib.compare_local_and_remote_file(host, local_file, file_to_check), \
            f'{file_to_check} has unexpected content'
E       AssertionError: /etc/pam.d/smartcard-auth has unexpected content
E       assert False
E        +  where False = <function compare_local_and_remote_file at 0x7f6ebcebfaf0>(<testinfra.host.Host paramiko://civ-vm-image-rhel-810-x86-64-main-9857133616-11036.westeurope.cloudapp.azure.com>, 'data/generic/smartcard-auth_rhel8', '/etc/pam.d/smartcard-auth')
E        +    where <function compare_local_and_remote_file at 0x7f6ebcebfaf0> = test_lib.compare_local_and_remote_file
test_suite/generic/test_generic.py:955: AssertionError
----------------------------- Captured stdout call -----------------------------
4c4,19
< auth sufficient pam_sss.so allow_missing_name
---
> auth        required      pam_env.so
> auth        [success=done ignore=ignore default=die] pam_pkcs11.so wait_for_card
> auth        required      pam_deny.so
> 
> account     required      pam_unix.so
> account     sufficient    pam_localuser.so
> account     sufficient    pam_succeed_if.so uid < 500 quiet
> account     required      pam_permit.so
> 
> password    optional      pam_pkcs11.so
> 
> session     optional      pam_keyinit.so revoke
> session     required      pam_limits.so
> -session     optional      pam_systemd.so
> session     [success=1 default=ignore] pam_succeed_if.so service in crond quiet use_uid
> session     required      pam_unix.so
\ No newline at end of file
=============================== warnings summary ===============================
test_suite/generic/test_generic.py: 41 warnings
test_suite/cloud/test_azure.py: 11 warnings
  /opt/app-root/lib64/python3.9/site-packages/pytest_html/fixtures.py:23: DeprecationWarning: The 'extra' fixture is deprecated and will be removed in a future release, use 'extras' instead.
    warnings.warn(
test_suite/generic/test_generic.py: 381 warnings
test_suite/cloud/test_azure.py: 123 warnings
  /opt/app-root/lib64/python3.9/site-packages/pytest_html/basereport.py:356: DeprecationWarning: The 'py' module is deprecated and support will be removed in a future release.
    warnings.warn(
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------- generated xml file: /tmp/report.xml ----------------------
--------------------------------- JSON report ----------------------------------
report saved to: /tmp/report.json
---------------- Generated html report: file:///tmp/report.html ----------------
=========================== short test summary info ============================
FAILED test_suite/generic/test_generic.py::TestsAuthConfig::test_smartcard_auth[paramiko://civ-vm-image-rhel-810-x86-64-main-9857133616-11036.westeurope.cloudapp.azure.com]
= 1 failed, 49 passed, 8 skipped, 12 deselected, 556 warnings in 518.39s (0:08:38) =
```

--- a/data/generic/smartcard-auth_rhel8.10
+++ b/data/generic/smartcard-auth_rhel8.10
@@ -1,0 +1,4 @@
+#%PAM-1.0
+# This file is auto-generated.
+# User changes will be destroyed the next time authselect is run.
+auth sufficient pam_sss.so allow_missing_name

--- a/test_suite/generic/test_generic.py
+++ b/test_suite/generic/test_generic.py
@@ -931,14 +931,19 @@ class TestsAuthConfig:
         """
         self.__check_pam_d_file_content(host, 'postlogin')
 
-    @pytest.mark.exclude_on(['>=rhel10.0', 'rhel8.10'])
+    @pytest.mark.exclude_on(['>=rhel10.0'])
     def test_smartcard_auth(self, host):
         """
         Check file /etc/pam.d/smartcard-auth
         Bugzilla: 1983683
         """
-
-        self.__check_pam_d_file_content(host, 'smartcard-auth')
+        if version.parse(host.system_info.release) == version.parse('8.10'):
+            local_file = 'data/generic/smartcard-auth_rhel8.10'
+            file_to_check = '/etc/pam.d/smartcard-auth'
+            assert test_lib.compare_local_and_remote_file(host, local_file, file_to_check), \
+                f'{file_to_check} has unexpected content'
+        else:
+            self.__check_pam_d_file_content(host, 'smartcard-auth')
 
     @pytest.mark.exclude_on(['>=rhel10.0'])
     def test_system_auth(self, host):

